lib/ansible/modules/cloud/amazon/lightsail_lb.py

```
@@ -0,0 +1,272 @@
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg='Error deleting load balancer {0}, error: {1}'.format(lb_name, e))
+
+    return (changed, loadbalancer)
+
+
+def core(module):
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg='region must be specified')
+
+    client = None
+    try:
+        client = boto3_conn(module, conn_type='client', resource='lightsail',
+                            region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
+        module.fail_json(msg='Failed while connecting to the lightsail service: %s' % e, exception=traceback.format_exc())
+
+    changed = False
+    state = module.params['state']
+    name = module.params['name']
+
+    if state == 'absent':
+        changed, lb_dict = delete_load_balancer(module, client, name)
+    elif state == 'present':
+        changed, lb_dict = create_load_balancer(module, client, name)
+
+    if lb_dict is None:
+        lb_dict = {}
+
+    module.exit_json(changed=changed, loadbalancer=camel_dict_to_snake_dict(lb_dict))
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        attachInstances=dict(type='list', default=[]),
+        port=dict(type='int', default=80),
+        healthCheckPath=dict(type='str', default=None)
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='Python module "boto3" is missing, please install it')
+
+    if not HAS_BOTOCORE:
+        module.fail_json(msg='Python module "botocore" is missing, please install it')
+
+    try:
+        core(module)
+    except (botocore.exceptions.ClientError, Exception) as e:
```

It's rarely appropriate to catch all Exceptions.

---

Do you have any suggestions?